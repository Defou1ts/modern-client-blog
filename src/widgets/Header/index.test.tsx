import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import { Header } from './index';

describe('Button', () => {
	it('renders a primary button', () => {
		render(<Header />);

		const titleText = screen.getByText('Modsen Client Blog');
		expect(titleText).toBeInTheDocument();
		expect(titleText).toBeVisible();

		const navtext1 = screen.getByText('Home');
		expect(navtext1).toBeInTheDocument();
		expect(navtext1).toBeVisible();

		const navtext2 = screen.getByText('Blog');
		expect(navtext2).toBeInTheDocument();
		expect(navtext2).toBeVisible();

		const navtext3 = screen.getByText('About Us');
		expect(navtext3).toBeInTheDocument();
		expect(navtext3).toBeVisible();

		const navtext4 = screen.getByText('Contact Us');
		expect(navtext4).toBeInTheDocument();
		expect(navtext4).toBeVisible();

		const buttonText = screen.getByText('Video about us');
		expect(buttonText).toBeInTheDocument();
		expect(buttonText).toBeVisible();
	});

	it('button header snapshot', () => {
		const component = renderer.create(<Header />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
