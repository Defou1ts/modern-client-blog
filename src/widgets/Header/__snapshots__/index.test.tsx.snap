// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button header snapshot 1`] = `
<header
  className="header"
>
  <div
    className="burger"
    onClick={[Function]}
  >
    <span />
  </div>
  <div
    className="headerWrapper"
  >
    <h1
      className="title"
    >
      Modsen Client Blog
    </h1>
    <nav
      className="nav"
    >
      <a
        className="navlink"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Home
      </a>
      <a
        className="navlink"
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Blog
      </a>
      <a
        className="navlink"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        About Us
      </a>
      <a
        className="navlink"
        href="/contact"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Contact Us
      </a>
    </nav>
    <button
      className="button secondary video"
    >
      Video about us
    </button>
  </div>
</header>
`;
